### ISSUE TITLE
EKS [request]: Support NLB Load balancing across multiple clusters

### ISSUE THREAD

**Issue by adhoot** [aws/containers-roadmap#1743]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
What do you want us to build?
NLB support for pods across multiple clusters

**Which service(s) is this request for?**
EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
What outcome are you trying to achieve, ultimately, and why is it hard/impossible to do right now? What is the impact of not having this problem solved? The more details you can provide, the better we'll be able to understand and solve the problem.
We would like the NLB to support sending traffic to the same workload running on multiple clusters. We are moving to this model of replicating our workload on multiple clusters for higher availability and reduce our RTO/RPO on cluster level failures. If we achieve this we can then treat clusters as cattle instead of pets and delete clusters at will for stateless applications using canary or blue green approach for clusters just like we do it for pods. Mesh already works across clusters but the Ingress path comes through AWS NLBs and the lack of this feature prevents this from working seamlessly.  

**Are you currently working around this issue?**
How are you currently solving this problem?
To workaround this we would have to reimplement the logic of the Ingress controller to maintain the targetgroupbinding between NLB and the backend pod IPs. The other workarounds for this problem involve using multiple NLBs, one per cluster with DNS across them or another proxy which are not ideal since they add other moving parts on the data path with their own limitations.  

**Additional context**
Anything else we should know?
The ideal design for this would allow us to create an NLB with a target selector that allows you to choose pods/services across clusters within the account. 

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by adhoot** [comment]
Yes DNS across NLBs is an option but it has the classic challenges of DNS with caching which means slower recovery and more customer impact on failures. With NLBs the latency of removing unhealthy targets (in this case an unhealthy cluster in a group) can be much faster and we can explicitly remove clusters instantaneously. See a related blog on this topic which is commonly cited and describes the same problem statement, https://aws.amazon.com/blogs/containers/onfidos-journey-to-a-multi-cluster-amazon-eks-architecture/.  It would be great for NLBs to support this out of the box.

**Comment by tschuett** [comment]
In this blog post, https://aws.amazon.com/blogs/containers/operating-a-multi-regional-stateless-application-using-amazon-eks/ they use the GlobelAccelerator as the external network component.

**Comment by daveortiz-ctct** [comment]
This is something we'd love to see too!  DNS based failover between clusters is definitely suboptimal because it makes failovers riskier than we'd like.

**Comment by aslatter** [comment]
I expect this is supported as of the AWS LB Controller 2.10: https://github.com/kubernetes-sigs/aws-load-balancer-controller/releases/tag/v2.10.0

The TGB CRD now supports "multiClusterTargetGroup: true", and services support the annotation `service.beta.kubernetes.io/aws-load-balancer-multi-cluster-target-group: "true"`.
